crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -1381,6 +1392,133 @@ pub fn decode_calldata(data: &Bytes) -> Result<(Vec<Fulfillment>, AssessorReceip
     ))
 }
 
+#[derive(Debug, Clone)]
+/// Represents the parameters for submitting a Merkle Root.
+pub struct Root {
+    /// The address of the set verifier contract.
+    pub verifier_address: Address,
+    /// The Merkle root of the proof.
+    pub root: B256,
+    /// The seal of the proof.
+    pub seal: Bytes,
+}
+
+#[derive(Debug, Clone)]
+/// Represents the parameters for pricing a request.
+pub struct Price {
+    /// The list of requests to be priced.
+    pub requests: Vec<ProofRequest>,
+    /// The list of client signatures for the requests.
+    pub client_sigs: Vec<Bytes>,
+}
+
+#[derive(Clone)]
+#[non_exhaustive]
+/// Builder for creating a fulfillment request.
+pub struct FulfillmentBuilder<P> {
```

Comment:
On the name, maybe we should call this something like `FulfillmentTx` or similar. I would call it `Fulfillment`, but that main is taken by the Solidity derived struct. It does follow a builder pattern, but it doesn't need to be called a "builder" because it does not help you create a `Fulfillment`, and has no `build` method.